REMAINING.PUBLIC.md

```diff
@@ -0,0 +1,62 @@
+### Remaining public aliases
+
+annotation_reader
+security.access.decision_manager
+security.acl.cache
+security.acl.dbal.connection
+security.acl.provider
+security.encoder_factory
+security.password_encoder
+session.handler
+session.storage
+templating.loader
+translator
+
+cache.app_clearer <- for the cache:pool:clear command to clear app pools
+$commandId <- for non-lazy commands which are get() at runtime
+router <- for the base controller
+templating <- for the base controller
+
+### Remaining public services
+
+file_locator
+filesystem
+kernel
+property_accessor
+property_info
+security.acl.dbal.schema
+security.authentication_utils
+test.client
+translation.extractor
+translation.loader
+translation.reader
+translation.writer
+translator
+validator
+
+cache.app <- for userland only
+cache_clearer <- for the cache:clear command
+cache.global_clearer <- for the cache:pool:clear command to clear all pools
+cache.system <- for userland only
+cache_warmer <- required to bootstrap the kernel
+data_collector.dump <- required to have dump() work very early when booting the kernel
+event_dispatcher <- required to wire console apps
+form.factory <- for the base controller
+http_kernel <- required to bootstrap the kernel
+profiler <- used in tests
+request_stack <- for the base controller
+routing.loader <- used by routing
+security.authorization_checker <- for the base controller
+security.csrf.token_manager <- for the base controller
+security.token_storage <- for the base controller
+serializer <- for the base controller
+session <- for the base controller
+state_machine.abstract <- state machines created by config are public
+workflow.abstract <- workflows created by config are public
```

Comment:
@lyrixx @Nyholm should we plan to make these services private in 4.0 and tell ppl that injecting workflow.registry is the way to go?

Reply:
No. The registry is only here for convenience in twig.
Users should inject the right workflow in other services. Adding an extra layer will make the core harder to understand and to test. It's exactly like doctrine. People should inject the right repository, not the whole Doctrine Manager.

Reply:
OK, then we can still make them private, isn't it?

Reply:
@lyrixx if you want them to inject it in the service, it can be private. Public is here for dynamic get.

But if they are meant to be the main services of the workflow component, they should indeed stay public for now.

Reply:
No, because we need to access it from a controller or a command.
Not everyone are using controller/command as a service.